lstchain/image/modifier.py

```diff
@@ -405,9 +408,7 @@ def calculate_noise_parameters(simtel_filename, data_dl1_filename,
     return extra_noise_in_dim_pixels, extra_bias_in_dim_pixels, \
         extra_noise_in_bright_pixels
 
-
 def calculate_required_additional_nsb(simtel_filename, data_dl1_filename, config=None):
```

Comment:
this function is quite big, I suggest extracting some parts with a given functionality in additional smaller functions

Reply:
Ok, I separated the real data part, which indeed will be useful to identify which (NSB-tuned) MC is needed to process a given data run